tuplex/codegen/src/IteratorContextProxy.cc

```diff
@@ -0,0 +1,494 @@
+#include <IteratorContextProxy.h>
+
+namespace tuplex {
+    namespace codegen {
+
+        SerializableValue IteratorContextProxy::initIterContext(LambdaFunctionBuilder &lfb, llvm::IRBuilder<> &builder,
+                                                            const python::Type &iterableType,
+                                                            const SerializableValue &iterable) {
+            using namespace llvm;
+
+            if(iterableType == python::Type::EMPTYLIST || iterableType == python::Type::EMPTYTUPLE) {
+                // use dummy value for empty iterator
+                return SerializableValue(_env->i64Const(0), _env->i64Const(8));
+            }
+
+            llvm::Type *iteratorContextType = _env->getIterIteratorType(iterableType);
+            auto initBBAddr = _env->getUpdateIteratorIndexFunction(builder, iterableType);
+            auto iteratorContextStruct = _env->CreateFirstBlockAlloca(builder, iteratorContextType, "iter_iterator_alloc");
+            llvm::Value *iterableStruct;
+            if(iterableType.isListType() || iterableType.isTupleType()) {
```

Comment:
dict type?

Reply:
Dictionary is handled through fallback.
Added check and error message.

Reply:
same here

Reply:
A check was added before this line in the updated file.